annotation-processing/src/main/java/com/exactpro/epfast/annotation/processing/FastErrorReporter.java

```diff
@@ -82,4 +84,54 @@ private void reportErrorMessage(String message, Stream<Element> elements) {
         errorRaised = true;
         elements.forEach(element -> messager.printMessage(Diagnostic.Kind.ERROR, message, element));
     }
+
+    public void reportDuplicatePackagesIfSuchExist(Collection<FastPackageElement> fastPackages) {
+        HashMap<String, ArrayList<NamedFastPackage>> packagesByName = new HashMap<>();
+        AnonymousFastPackage anonymousFastPackage = null;
+
+        for (FastPackageElement fastPackage : fastPackages) {
+            if (fastPackage instanceof NamedFastPackage) {
+                packagesByName.computeIfAbsent(fastPackage.getName(), key -> new ArrayList<>())
+                    .add((NamedFastPackage) fastPackage);
+            } else {
+                anonymousFastPackage = (AnonymousFastPackage) fastPackage;
+            }
+        }
+        packagesByName.values().stream()
+            .filter(packages -> packages.size() > 1)
+            .forEach(this::reportDuplicatePackages);
+
+        if (anonymousFastPackage != null) {
+            ArrayList<NamedFastPackage> namedPackagesForRootName = packagesByName.get(anonymousFastPackage.getName());
+            if (namedPackagesForRootName != null) {
+                reportAnonymousPackageEqualsNamedPackage(namedPackagesForRootName);
+            }
+        }
+
+    }
+
+    public void reportInvalidPackagesIfSuchExist(Stream<FastPackageElement> fastPackages) {
+        fastPackages.filter(entry -> !entry.isValidPackage())
```

Comment:
Check (filtering) should be done if FastProcessor. Here only output should remain